src/com/jwetherell/algorithms/strings/StringFunctions.java

```diff
@@ -13,20 +13,29 @@ public class StringFunctions {
     private static final char SPACE = ' ';
 
     public static final String reverseWithStringConcat(String string) {
-        String output = new String();
-        for (int i = (string.length() - 1); i >= 0; i--) {
-            output += (string.charAt(i));
-        }
-        return output;
+    if (string == null || string.isEmpty()) {
+        return ""; // Return empty string for null or empty input
     }
 
-    public static final String reverseWithStringBuilder(String string) {
-        final StringBuilder builder = new StringBuilder();
-        for (int i = (string.length() - 1); i >= 0; i--) {
-            builder.append(string.charAt(i));
-        }
-        return builder.toString();
+    String output = new String();
+    for (int i = string.length() - 1; i >= 0; i--) {
+        output += string.charAt(i);
     }
+    return output;
+}
+
+public static final String reverseWithStringBuilder(String string) {
+    if (string == null || string.isEmpty()) {
```

Comment:
Check for null or empty input and return an empty string if true. This prevents null pointer exceptions and unnecessary processing